plugins/pulp_rpm/plugins/importers/yum/pulp_solv.py

```diff
@@ -722,12 +735,20 @@ def load(self):
         if self._loaded:
             return
 
-        self.load_repo(self.source_repo.repo_id)
-        if self.target_repo:
-            target_repo_id = self.target_repo.repo_id
+        # Load the primary source and (if provided) target repos
+        self.load_source_repo(self.primary_source_repo.repo_id)
+        if self.primary_target_repo:
```

Comment:
Can it be `None`? If yes, what's next? It seems like we proceed further even if it's not set. Is it intended?

Reply:
I guess it depends how defensive we want to be, but I don't think Pulp will ever pass in a None

Reply:
I'm still unsure why we allow to continue without a primary_target_repo. Should the operaiton fail if there is no primary_target_repo?

Reply:
I think "(relaxed) recursive mode" doesn't require you to feed the solver a target repo